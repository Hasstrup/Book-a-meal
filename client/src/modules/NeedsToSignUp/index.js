import React, { Component, Fragment } from 'react';
import { connect } from 'react-redux';
import { withRouter } from 'react-router-dom';
import { SignUpUser, LogInUser } from '../../actions/users';
import { DispatchNotification } from '../../actionTypes/misc';
import './index.scss';

const SignUpComponent = ({ renderLogin, handleClick, handleSubmit, handleChange }) => (
  <div className="sign-up-user-form">
    <div className="render-form-single">
      <label htmlFor="email" className="render-form-label"> Email </label>
      <input onChange={handleChange} type="email" required name="email" className="render-form-input" />
    </div>
    <div className="render-form-single">
      <label htmlFor="password" > Password </label>
      <input onChange={handleChange} type="password" required className="render-form-input" name="password" />
    </div>
    { !renderLogin &&
    <Fragment>
      <div className="render-form-single">
        <label htmlFor="name" className="render-form-label"> Full Name</label>
        <input onChange={handleChange} type="text" required className="render-form-input" name="name" />
      </div>
      <div className="render-form-single">
        <label htmlFor="username" className="render-form-label"> Username </label>
        <input onChange={handleChange} required className="render-form-input" name="username" />
      </div>
    </Fragment>
    }
    <button onClick={handleSubmit}> Submit </button>
    <p className="__user_already" onClick={handleClick}>{ renderLogin ? 'Create new account' : 'I already have an account '}</p>
  </div>
);

class AuthenticateUserForm extends Component {
   state = { renderLogin: false, preamble: "Holla. Let's get you started" };
    handleChange = e => this.setState({ [e.target.getAttribute('name')]: e.target.value });

    
    handleSubmit = () => {
      const { email, password, username, name } = this.state;
      const filterThrough = obj => Object.values(obj).filter(item => !item || !item.length).length;
      if (this.state.renderLogin && !filterThrough({ email, password })) return this.props.dispatch(LogInUser({ email, password })(this.props.history));
      if (!filterThrough({ name, password, username, email })) return this.props.dispatch(SignUpUser({ email, password, firstname: name.split(' ')[0], username })(this.props.history));
      this.props.dispatch(DispatchNotification('Hey, please fill in the required fields'));
    }

    render = () => (
      <div className="render-auth-modal" style={{ display: this.props.requiresPermission ? 'flex' : 'none' }}>
        <div className="auth-modal-backdrop" />
        <div className="auth-modal-main">
          <div className="__auth-modal-house">
            <p className="auth-preamble">{this.state.preamble}</p>
            <p className="auth-preamble-text">{this.props.preamble || ''}</p>
          </div>
          <div className="__auth-modal-main">
            <SignUpComponent
              handleClick={() => this.setState({ renderLogin: !this.state.renderLogin, preamble: (this.state.renderLogin ? "Holla. Let's get you started" : 'Hungry again? Nice.') })}
              renderLogin={this.state.renderLogin}
              handleChange={this.handleChange}
              handleSubmit={this.handleSubmit}
            />
          </div>
        </div>
      </div>
    )
}

const mapStateToProps = state => ({
    requiresPermission: state.users.requiresPermission
})

export default connect(mapStateToProps)(withRouter(AuthenticateUserForm));

