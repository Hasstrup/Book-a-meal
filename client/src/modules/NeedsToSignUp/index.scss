.render-auth-modal {
    position: fixed;
    top: 0;
    left: 0;
    width: 100%;
    height: 100%;
    z-index: 1000;
    display: flex;
    background-color: rgba(253, 254, 254, 0.9);
    .auth-modal-backdrop {
        width: 100%;
        height: 100%;
        top: 0;
        left: 0;
    }
    .auth-modal-main {
        display: flex; 
        height: 70%;
        width: 60%;
        background-color: white;
        border-radius: 5px;
        align-self: center;
        margin: 0 auto;
        position: relative; 
        right: 35%;
        box-shadow: 1px 1px 10px 1px #F2F3F4;
        flex-flow: column nowrap;
        align-items: center;
        .__auth-modal-house {
            height: 10%;
            width: 100%;
            .auth-preamble {
                color: #5A5A5A;
                text-align: center;
                font: {
                    size: 25px;
                    weight: bold;
                    family: gotham;
                }
            }
        }
        .__auth-modal-main {
            height: 90%;
            width: 100%;
            display: flex; 
            flex-flow: column nowrap;
            align-items: center;
            .sign-up-user-form {
                display: flex;
                height: 100%;
                width: 50%;
                align-self: center;
                flex-flow: column nowrap;
                justify-content: space-evenly;
                .__user_already {
                    font-size: 13px; 
                    font-family: apercu;
                    color: #BDC3C7;
                    text-align: center;
                    &:hover {
                        cursor: pointer;
                        text-decoration: underline;
                    }
                }
                .render-form-single {
                    .render-form-input {
                        display: block;
                        height: 40px;
                        width: 100%;
                        border: none;
                        border-bottom: 0.5px solid #E5E7E9;
                        font-family: apercu;
                        font-size: 14px;
                        &:focus {
                            outline: none;
                        }
                    }
                    label {
                        color: #BDC3C7;
                    }
                    font: {
                        size: 16px;
                        family: apercu;
                    }
                }
                button {
                    background-color: #ff183f;
                    color: white;
                    font-family: gotham;
                    font-size: 18px;
                    height: 50px !important; 
                    width: 290px;
                    border: none;
                    border-radius: 3px;
                    &:hover {
                        cursor: pointer;
                    }
                }
            }
        }
    }
}