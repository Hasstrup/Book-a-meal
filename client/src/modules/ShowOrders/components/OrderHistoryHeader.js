import React from 'react';

export default () => (
  <div className="orders-page-side-one">
      <div className="orders-heading-and-desc">
        <p className="orders-page-heading"> Orders </p>
        <p className="orders-page-desc">
          This is a page dedicated for you
          to manage your orders as a regular
          user or as a vendor. You can check your
          monitor the current items in cart
          and check with your previous order history.
          Enjoy!
        </p>
      </div>
      <div className="convey-account-details">
        <p className="convey-account-order">
           You are currently viewing this page as <br />
          <span id="user-details-init"> Hasstrup Ezekiel </span> <br />
        </p>
        <div className="display-click-message">
          <p onClick="showAccountsMenu()"> Click to switch account </p>
          <div className="display-current-user">
            <div className="user-list-item"> Hello Desi kitchen </div>
            <div className="user-list-item"> Hasstrup Ezekiel </div>
          </div>
        </div>
      </div>
    </div>
);
