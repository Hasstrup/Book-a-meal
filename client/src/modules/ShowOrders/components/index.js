import CurrentOrder from './CurrentOrder';
import OrderHistoryGrid from './OrderHistoryGrid';
import { OrderHistoryComponent as RenderMainHistory } from './RenderMainHistory';

export const ComponentList = [
  CurrentOrder,
  OrderHistoryGrid,
  RenderMainHistory
];

export const PropList = [
  { meals: [], history: () => {} },
  { orders: [], title: '' },
  { orders: [] }
];

export const NameList = [
  'Current Order',
  'OrderHistoryGrid',
  'RenderMainHistory'
];
