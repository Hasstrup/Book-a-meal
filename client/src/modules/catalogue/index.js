import React, { Component } from 'react';
import { connect } from 'react-redux';
import PropTypes from 'prop-types';
import '../styles/catalogue.scss';
import { MainCatalogue } from '../../mixins/cards/singleMenu';
import { FetchCatalogue } from '../../actions/menus/';


class Catalogue extends Component {
  static propTypes = {
    dispatch: PropTypes.func.isRequired,
    menus: PropTypes.arrayOf(PropTypes.object).isRequired,
    history: PropTypes.object.isRequired
  }

  componentDidMount = () => {
    if (!this.props.menus) return this.fetchMenus();
  }

  fetchMenus = () => this.props.dispatch(FetchCatalogue(this.props.history));

  render = () => (
    <div className="main-body">
      <CatalogueFirstRow />
      <Selectors />
      <CatalogueGridMain history={this.props.history} menus={this.props.menus || []} />
    </div>
  );
}

/*= ========== Components ============================ */

const CatalogueFirstRow = () => (
  <div className="introductory-grid">
    <div className="side-one-IT">
      <div className="side-one-child-1">
        <p className="side-one-IT-cat"> Catalogue </p>
        <div className="side-one-trending"> Today</div>
      </div>
      <p className="side-one-IT-option"> <span> 237 </span> Menus </p>
      <p className="side-one-IT-option"> <span> 1034</span> Meal options </p>
      <p className="side-one-IT-option"> <span> 101 </span> Vendors</p>
    </div>

    <div className="side-two-IT">
      <p className="side-two-IT-info">
                Sharing a menu is super easy! There's no restrictions
                whatsoever, all you have to do set up your kitchen/catering
                service, add some meal options, set up a menu and we're good
                to go .
      </p>
      <p className="side-two-IT-info" id="side-two-share-button"> Share your menu now</p>
    </div>
  </div>
);


const Selectors = () => (
  <div className="options-grid">
    <p> Top </p>
    <p> Featured </p>
    <p> All </p>
    <p> Favourites </p>
    <p> Promote </p>
  </div>
);


const CatalogueGridMain = props => (
  <div className="main-items-grid">
    { /* remember to get some mock || main data to fill in this place */ }
    { MainCatalogue(props.menus)(props) }
  </div>
);

const mapStateToProps = state => ({
  menus: state.menus.catalog
});
// these should be refactored after the application is completed right
export default connect(mapStateToProps)(Catalogue);

