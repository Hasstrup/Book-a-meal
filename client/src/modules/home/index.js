import React, { Component } from 'react';
import { connect } from 'react-redux';
import '../styles/index.scss';
import AnimatingBannerComponent from './components/banner';
import ImageFloatingComponent from './components/floater';
import LoginFormComponent from './components/auth';
import { SignUpUser, LogInUser, GetLoggedInUser } from '../../actions/users/';
/* eslint-disable */


export class HomeContainer extends Component {
  constructor(props) {
    super(props);
    if(this.props.user) return this.props.history.push('/catalogue');
    
  }
  componentDidMount = () => {
    GetLoggedInUser();
  }
  
  render = () => (<HomeComponent history={this.props.history} dispatch={this.props.dispatch} />)
}

export const HomeComponent = ({ history, dispatch }) => (
  <div>
  <AnimatingBannerComponent />
  <LoginFormComponent logInUser={(body) => dispatch(LogInUser(body)(history))} createUser={(body) => dispatch(SignUpUser(body)(history))}/>
  </div>
)

const mapStateToProps = (state) => ({
    user: state.users.current
})

export default connect(mapStateToProps)(HomeContainer);
