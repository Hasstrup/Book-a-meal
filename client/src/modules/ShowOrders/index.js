import React, { Component } from 'react';
import { connect } from 'react-redux';
import swal from 'sweetalert';
import CurrentOrderComponent from './components/CurrentOrder';
import RenderActualHistory from './components/RenderMainHistory';
import OrderHistoryHeader from './components/OrderHistoryHeader';
import AuthOnly from '../../hocs/AuthOnly';
import { DispatchNotification } from '../../actionTypes/misc';
import { fetchAllOrders } from '../../actions/orders';
import { CartOps } from '../../actions/ops/cart';
import '../styles/orders.scss';


class OrderHistoryPage extends Component {
  constructor(props) {
    super(props);
    this.state = {};
    this.state.target = this.props.user;
    this.state.ordersInView = this.props.orders[this.props.user.id];
    this.state.currentMealsInCart = CartOps()()({});
  }

    componentDidMount = () => {
      const { dispatch, user, kitchen } = this.props;
      /* eslint-disable-next-line  */
      user && dispatch(fetchAllOrders('user')(user.id));
      kitchen && dispatch(fetchAllOrders('kitchen')(kitchen.id));
    }

    componentDidUpdate = (prevProps) => {
      const { orders } = this.props;
      const { target } = this.state;
      if (orders[target.id] && orders[target.id].length) {
        if (!prevProps.orders[target.id] || !prevProps.orders[target.id].length) return this.setState({ ordersInView: this.props.orders[this.state.target.id] });
      }
    }

    changeTarget = (target) => {
      const { dispatch, orders } = this.props;
      dispatch(DispatchNotification(`You are now seeing orders made ${target.name ? 'to' : 'by'} ${target.name || target.firstname}`));
      dispatch(fetchAllOrders(target.type)(target.id));
      this.state.target = target;
      // this rerender here will update the dom;
      this.setState({ ordersInView: orders[`${target.id}`] });
    }

    render = () => (
      <div className="main-body-component" style={{ minHeight: '100vh' }}>
        <OrderHistoryHeader kitchen={this.props.kitchen} handleChange={this.changeTarget} user={this.props.user} target={this.state.target} />
        <div className="orders-page-side-one-main">
          { this.state.currentMealsInCart && this.state.currentMealsInCart.length && <CurrentOrderComponent meals={this.state.currentMealsInCart} history={this.props.history} /> }
          <RenderActualHistory
            orders={this.state.ordersInView}
            determineRenderBy={this.state.ordersInView}
            callback={() => {
            this.props.history.push('/catalogue');
            swal(`Here you go ${this.props.user.firstname}, make your from here`, 'PS: they are all yummy', 'success');
          }}
            text={this.state.target.id === this.props.user.id ? ` Aw schucks, ${this.state.target.firstname} hasn't made an order yet` : 'Looks like no one has made an order yet'}
            subtitle="order your first item"
            emptyContainerStyle={{
            height: '10%',
            position: 'relative',
            bottom: '-14vh'
          }}
          />
        </div>
      </div>
    )
}

const mapStateToProps = state => ({
  kitchen: state.kitchens.target,
  user: state.users.current,
  orders: state.orders.allOrders || {}
});

export default connect(mapStateToProps)(AuthOnly(OrderHistoryPage));
