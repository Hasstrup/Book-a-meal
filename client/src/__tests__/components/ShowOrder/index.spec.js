import React from 'react';
import { shallow } from 'enzyme';
import { stub, spy } from 'sinon';
import defaultProps from '../../helpers';
import { OrderHistoryPage } from '../../../modules/ShowOrders';
import RenderActualHistory from '../../../modules/ShowOrders/components/RenderMainHistory';
import OrderHistoryHeader from '../../../modules/ShowOrders/components/OrderHistoryHeader';

let wrapper;
const dispatchSpy = spy(defaultProps, 'dispatch');

const menu = {
  name: 'Test Menu',
  description: 'This is a test menu',
  kitchen: {
    name: 'This is great',
    description: 'This is my test menu',
    id: 3,
    MenuofTheDay: null
  },
  meals: [{
    name: 'This is a menu',
    description: '',
    price: 5000,
    kitchenId: 2
  }]
};

const orders = {
  1: [{ meal: 1 }]
};

const user = {
  firstname: 'Hasstru',
  lastname: 'Ezekiel',
  email: 'hass@mail.com',
  kitchen: menu.kitchen,
  id: 1
};

describe('Order History Page', () => {
  it('should match snapshot on shallow render', () => {
    wrapper = shallow(<OrderHistoryPage orders={orders} {...defaultProps} user={user} kitchen={menu.kitchen} meals={menu.meals} />);
    expect(wrapper).toMatchSnapshot();
  });

  it('should contain two children on shallow render', () => {
    expect(wrapper.find(RenderActualHistory)).toHaveLength(1);
    expect(wrapper.find(OrderHistoryHeader)).toHaveLength(1);
  });

  describe('Order History methods', () => {
    it('Change target should change the orders currently in view', () => {
      const { changeTarget, state } = wrapper.instance();
      changeTarget(user);
      expect(dispatchSpy.called).toEqual(true);
      expect(state.target.id).toEqual(1);
      expect(state.ordersInView[0].meal).toEqual(1);
    });
  });

  it('should render successfully on deep mounting', () => {
    expect(wrapper).toMatchSnapshot();
  });
});
