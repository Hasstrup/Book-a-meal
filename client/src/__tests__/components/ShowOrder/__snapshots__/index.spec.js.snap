// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Order History Page should match snapshot on shallow render 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderHistoryPage
    dispatch={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    kitchen={
      Object {
        "MenuofTheDay": null,
        "description": "This is my test menu",
        "id": 3,
        "name": "This is great",
      }
    }
    meals={
      Array [
        Object {
          "description": "",
          "kitchenId": 2,
          "name": "This is a menu",
          "price": 5000,
        },
      ]
    }
    orders={
      Object {
        "1": Array [
          Object {
            "meal": 1,
          },
        ],
      }
    }
    user={
      Object {
        "email": "hass@mail.com",
        "firstname": "Hasstru",
        "id": 1,
        "kitchen": Object {
          "MenuofTheDay": null,
          "description": "This is my test menu",
          "id": 3,
          "name": "This is great",
        },
        "lastname": "Ezekiel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          handleChange={[Function]}
          kitchen={
            Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            }
          }
          target={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
          user={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
        />,
        <div
          className="orders-page-side-one-main"
        >
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                Object {
                  "meal": 1,
                },
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-14vh",
                "height": "10%",
                "position": "relative",
              }
            }
            orders={
              Array [
                Object {
                  "meal": 1,
                },
              ]
            }
            subtitle="order your first item"
            text=" Aw schucks, Hasstru hasn't made an order yet"
          />
        </div>,
      ],
      "className": "main-body-component",
      "style": Object {
        "minHeight": "100vh",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [Function],
          "kitchen": Object {
            "MenuofTheDay": null,
            "description": "This is my test menu",
            "id": 3,
            "name": "This is great",
          },
          "target": Object {
            "email": "hass@mail.com",
            "firstname": "Hasstru",
            "id": 1,
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "lastname": "Ezekiel",
          },
          "user": Object {
            "email": "hass@mail.com",
            "firstname": "Hasstru",
            "id": 1,
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "lastname": "Ezekiel",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                }
              }
              orders={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              subtitle="order your first item"
              text=" Aw schucks, Hasstru hasn't made an order yet"
            />,
          ],
          "className": "orders-page-side-one-main",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "callback": [Function],
              "determineRenderBy": Array [
                Object {
                  "meal": 1,
                },
              ],
              "emptyContainerStyle": Object {
                "bottom": "-14vh",
                "height": "10%",
                "position": "relative",
              },
              "orders": Array [
                Object {
                  "meal": 1,
                },
              ],
              "subtitle": "order your first item",
              "text": " Aw schucks, Hasstru hasn't made an order yet",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            handleChange={[Function]}
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            target={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "id": 1,
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "id": 1,
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />,
          <div
            className="orders-page-side-one-main"
          >
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                }
              }
              orders={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              subtitle="order your first item"
              text=" Aw schucks, Hasstru hasn't made an order yet"
            />
          </div>,
        ],
        "className": "main-body-component",
        "style": Object {
          "minHeight": "100vh",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "target": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
            "user": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <Unknown
                callback={[Function]}
                determineRenderBy={
                  Array [
                    Object {
                      "meal": 1,
                    },
                  ]
                }
                emptyContainerStyle={
                  Object {
                    "bottom": "-14vh",
                    "height": "10%",
                    "position": "relative",
                  }
                }
                orders={
                  Array [
                    Object {
                      "meal": 1,
                    },
                  ]
                }
                subtitle="order your first item"
                text=" Aw schucks, Hasstru hasn't made an order yet"
              />,
            ],
            "className": "orders-page-side-one-main",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "callback": [Function],
                "determineRenderBy": Array [
                  Object {
                    "meal": 1,
                  },
                ],
                "emptyContainerStyle": Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                },
                "orders": Array [
                  Object {
                    "meal": 1,
                  },
                ],
                "subtitle": "order your first item",
                "text": " Aw schucks, Hasstru hasn't made an order yet",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;

exports[`Order History Page should render successfully on deep mounting 1`] = `
ShallowWrapper {
  "length": 1,
  Symbol(enzyme.__root__): [Circular],
  Symbol(enzyme.__unrendered__): <OrderHistoryPage
    dispatch={[Function]}
    history={
      Object {
        "push": [Function],
      }
    }
    kitchen={
      Object {
        "MenuofTheDay": null,
        "description": "This is my test menu",
        "id": 3,
        "name": "This is great",
      }
    }
    meals={
      Array [
        Object {
          "description": "",
          "kitchenId": 2,
          "name": "This is a menu",
          "price": 5000,
        },
      ]
    }
    orders={
      Object {
        "1": Array [
          Object {
            "meal": 1,
          },
        ],
      }
    }
    user={
      Object {
        "email": "hass@mail.com",
        "firstname": "Hasstru",
        "id": 1,
        "kitchen": Object {
          "MenuofTheDay": null,
          "description": "This is my test menu",
          "id": 3,
          "name": "This is great",
        },
        "lastname": "Ezekiel",
      }
    }
  />,
  Symbol(enzyme.__renderer__): Object {
    "batchedUpdates": [Function],
    "getNode": [Function],
    "render": [Function],
    "simulateEvent": [Function],
    "unmount": [Function],
  },
  Symbol(enzyme.__node__): Object {
    "instance": null,
    "key": undefined,
    "nodeType": "host",
    "props": Object {
      "children": Array [
        <Unknown
          handleChange={[Function]}
          kitchen={
            Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            }
          }
          target={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
          user={
            Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            }
          }
        />,
        <div
          className="orders-page-side-one-main"
        >
          <Unknown
            callback={[Function]}
            determineRenderBy={
              Array [
                Object {
                  "meal": 1,
                },
              ]
            }
            emptyContainerStyle={
              Object {
                "bottom": "-14vh",
                "height": "10%",
                "position": "relative",
              }
            }
            orders={
              Array [
                Object {
                  "meal": 1,
                },
              ]
            }
            subtitle="order your first item"
            text=" Aw schucks, Hasstru hasn't made an order yet"
          />
        </div>,
      ],
      "className": "main-body-component",
      "style": Object {
        "minHeight": "100vh",
      },
    },
    "ref": null,
    "rendered": Array [
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "function",
        "props": Object {
          "handleChange": [Function],
          "kitchen": Object {
            "MenuofTheDay": null,
            "description": "This is my test menu",
            "id": 3,
            "name": "This is great",
          },
          "target": Object {
            "email": "hass@mail.com",
            "firstname": "Hasstru",
            "id": 1,
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "lastname": "Ezekiel",
          },
          "user": Object {
            "email": "hass@mail.com",
            "firstname": "Hasstru",
            "id": 1,
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "lastname": "Ezekiel",
          },
        },
        "ref": null,
        "rendered": null,
        "type": [Function],
      },
      Object {
        "instance": null,
        "key": undefined,
        "nodeType": "host",
        "props": Object {
          "children": Array [
            undefined,
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                }
              }
              orders={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              subtitle="order your first item"
              text=" Aw schucks, Hasstru hasn't made an order yet"
            />,
          ],
          "className": "orders-page-side-one-main",
        },
        "ref": null,
        "rendered": Array [
          undefined,
          Object {
            "instance": null,
            "key": undefined,
            "nodeType": "function",
            "props": Object {
              "callback": [Function],
              "determineRenderBy": Array [
                Object {
                  "meal": 1,
                },
              ],
              "emptyContainerStyle": Object {
                "bottom": "-14vh",
                "height": "10%",
                "position": "relative",
              },
              "orders": Array [
                Object {
                  "meal": 1,
                },
              ],
              "subtitle": "order your first item",
              "text": " Aw schucks, Hasstru hasn't made an order yet",
            },
            "ref": null,
            "rendered": null,
            "type": [Function],
          },
        ],
        "type": "div",
      },
    ],
    "type": "div",
  },
  Symbol(enzyme.__nodes__): Array [
    Object {
      "instance": null,
      "key": undefined,
      "nodeType": "host",
      "props": Object {
        "children": Array [
          <Unknown
            handleChange={[Function]}
            kitchen={
              Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              }
            }
            target={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "id": 1,
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
            user={
              Object {
                "email": "hass@mail.com",
                "firstname": "Hasstru",
                "id": 1,
                "kitchen": Object {
                  "MenuofTheDay": null,
                  "description": "This is my test menu",
                  "id": 3,
                  "name": "This is great",
                },
                "lastname": "Ezekiel",
              }
            }
          />,
          <div
            className="orders-page-side-one-main"
          >
            <Unknown
              callback={[Function]}
              determineRenderBy={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              emptyContainerStyle={
                Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                }
              }
              orders={
                Array [
                  Object {
                    "meal": 1,
                  },
                ]
              }
              subtitle="order your first item"
              text=" Aw schucks, Hasstru hasn't made an order yet"
            />
          </div>,
        ],
        "className": "main-body-component",
        "style": Object {
          "minHeight": "100vh",
        },
      },
      "ref": null,
      "rendered": Array [
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "function",
          "props": Object {
            "handleChange": [Function],
            "kitchen": Object {
              "MenuofTheDay": null,
              "description": "This is my test menu",
              "id": 3,
              "name": "This is great",
            },
            "target": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
            "user": Object {
              "email": "hass@mail.com",
              "firstname": "Hasstru",
              "id": 1,
              "kitchen": Object {
                "MenuofTheDay": null,
                "description": "This is my test menu",
                "id": 3,
                "name": "This is great",
              },
              "lastname": "Ezekiel",
            },
          },
          "ref": null,
          "rendered": null,
          "type": [Function],
        },
        Object {
          "instance": null,
          "key": undefined,
          "nodeType": "host",
          "props": Object {
            "children": Array [
              undefined,
              <Unknown
                callback={[Function]}
                determineRenderBy={
                  Array [
                    Object {
                      "meal": 1,
                    },
                  ]
                }
                emptyContainerStyle={
                  Object {
                    "bottom": "-14vh",
                    "height": "10%",
                    "position": "relative",
                  }
                }
                orders={
                  Array [
                    Object {
                      "meal": 1,
                    },
                  ]
                }
                subtitle="order your first item"
                text=" Aw schucks, Hasstru hasn't made an order yet"
              />,
            ],
            "className": "orders-page-side-one-main",
          },
          "ref": null,
          "rendered": Array [
            undefined,
            Object {
              "instance": null,
              "key": undefined,
              "nodeType": "function",
              "props": Object {
                "callback": [Function],
                "determineRenderBy": Array [
                  Object {
                    "meal": 1,
                  },
                ],
                "emptyContainerStyle": Object {
                  "bottom": "-14vh",
                  "height": "10%",
                  "position": "relative",
                },
                "orders": Array [
                  Object {
                    "meal": 1,
                  },
                ],
                "subtitle": "order your first item",
                "text": " Aw schucks, Hasstru hasn't made an order yet",
              },
              "ref": null,
              "rendered": null,
              "type": [Function],
            },
          ],
          "type": "div",
        },
      ],
      "type": "div",
    },
  ],
  Symbol(enzyme.__options__): Object {
    "adapter": ReactSixteenAdapter {
      "options": Object {
        "enableComponentDidUpdateOnSetState": true,
      },
    },
  },
}
`;
